www/source/blog/2017-11-16-angular-seed.html.md

```diff
@@ -0,0 +1,117 @@
+---
+title: NodeJS Scaffolding with Angular Seed!
+date: 2017-11-16
+author: Nell Shamrell-Harrington
+tags: Scaffolding, Node, Angular
+category: community
+classes: body-article
+---
+
+Hello my fellow Habicats!
+
+Do you have legacy Angular projects that use [angular-seed](https://github.com/mgechev/angular-seed)? Now you can package them with Habitat using the Node Scaffolding! Let's give it a go!
+
+## Pre-requisites
+* A Github Account
+* NodeJS installed on your workstation
+* Angular installed on your workstation
+* Habitat installed on your workstation
+* Docker installed and running on your workstation
+* Text editor of your choice (in this post I use vim, but feel free to substitute your preferred editor)
+
+## Setting up your application
+
+Go ahead and clone the source code for [angular-seed](https://github.com/mgechev/angular-seed).
+
+```console
+$ git clone https://github.com/mgechev/angular-seed
+$ cd angular-seed
+```
+
+Now install the application's dependencies - in this example we use npm, but you can use yarn if you like.
+
+```console
+$ npm install
+```
+
+Angular-seed comes with both a package-lock.json and yarn.lock file. We can only use one package manager with the Node Scaffolding. If you are using npm, delete yarn.lock. If you are using yarn, delete package-lock.json.
+
+```console
+$ rm yarn.lock
+```
+
+If you'd like, go ahead and run the application locally:
+
+```console
+$ npm start
+```
+
+And you should be able to access the app at http://localhost:5555.
+
+## Code changes
+
+There are two code changes we need to make to angular-seed in order to use it with Habitat. Open up the package.json file and downgrade the following dependencies to the shown versions:
+
+```console
+"typescript": "2.4.2",
+"@compodoc/compodoc": "1.0.3",
```

Comment:
Why is it necessary to remove these?

Reply:
Dependency hell is the short version - several of the dependencies in the Angular Seed framework use typescript 2.4.2, but one requires typescript 2.6.0 (@compodoc/compodoc) - I have not been able to get the node scaffolding to accept two different versions of that dependency (and discovered that there are very strong opinions in the Node community around allowing a package.json to require multiple versions of a dependency).  I've gone pretty deep into peer dependency land trying to solve this (and barely found my way out). Until all dependencies in Angular Seed require the same version of typescript, this seemed like the most straightforward solution. 

I'll have to think on a good way to explain this...

Reply:
> Until all dependencies in Angular Seed require the same version of typescript, this seemed like the most straightforward solution.

Ah, this is definitely going to be common, then. But I guess still don't understand why, if you can run  `npm run build.prod` for example (which will ultimately resolve all dependencies, including whatever versions of typescript might be conflicting), we can't do the same with scaffolding.

Reply:
I will add that as an issue to core-plans and mention it in the post.

Reply:
https://github.com/habitat-sh/core-plans/issues/979